bundles/core/src/test/java/com/adobe/cq/wcm/core/components/internal/link/DefaultPathProcessorTest.java

```diff
@@ -74,5 +77,27 @@ void testSanitizeInternalLik() {
         assertEquals("/some%20space#internal", underTest.sanitize(path, request));
     }
 
+    @Test
+    void testVanityUrl() {
+        Page page = context.create().page("/content/links/site1/en", "/conf/example",
+                ImmutableMap.of("sling:vanityPath", "vanity.html"));
+        DefaultPathProcessor underTest = context.registerInjectActivateService(new DefaultPathProcessor(), ImmutableMap.of(
+                "vanityConfig", DefaultPathProcessor.VanityConfig.MAPPING.getValue()));
+        assertTrue(underTest.accepts(page.getPath() + LinkHandler.HTML_EXTENSION, context.request()));
+        assertEquals("/vanity.html", underTest.map(page.getPath() + LinkHandler.HTML_EXTENSION, context.request()));
```

Comment:
I would also add an assert here for `underTest.sanitize()` and `underTest.externalize()`.

Reply:
addressed in ec9353b6aa3a978c24955c37f191ef1990869744